lib/cn/csched_sp3.c

```diff
@@ -1256,11 +1276,32 @@ sp3_prune_trees(struct sp3 *sp)
  * and is executed on an external thread.
  */
 static void
-sp3_work_complete(struct cn_compaction_work *w)
+sp3_work_checkpoint(struct cn_compaction_work *w)
+{
+    sp3_enqueue_dirty_node(w, w->cw_output_nodev[0]);
+
+    /* [HSE_REVISIT] Don't wake up the scheduler after each call.
+     */
+    //sp3_monitor_wake(w->cw_sched);
+}
```

Comment:
I think the dirty list should be tied to the sp3 struct to reduce the overhead of looking for dirty nodes (see how `sp3_process_worklist()` process the worklist).  I would argue the dirty node list should be processed as frequently as the worklist since they're both notifications to the scheduler that a compaction job has modified the tree.